src/main/java/appeng/me/storage/NetworkInventoryHandler.java

```diff
@@ -94,24 +101,11 @@ public T injectItems(T input, final Actionable type, final BaseActionSource src)
         final int size = priorityInventory.size();
 
         int i = 0;
-        boolean stickyInventoryFound = false;
-        // Try to insert into all sticky inventories which are at the beginning of the list
-        for (; i < size && input != null; i++) {
-            final IMEInventoryHandler<T> inv = priorityInventory.get(i);
-            if (!inv.getSticky()) break;
-
-            if (inv.canAccept(input)
-                    && (inv.isPrioritized(input) || inv.extractItems(input, Actionable.SIMULATE, src) != null)) {
-                input = inv.injectItems(input, type, src);
-                stickyInventoryFound = true;
-            }
-        }
-
-        if (stickyInventoryFound || input == null || i >= size) {
```

Comment:
Can we not just keep this loop up here? Moving it down just complicates the other loop and also causes it to waste performance by running pass 2 for sticky inventories even though that does nothing.

I think this could stay as is by changing the exit check to `if (!inv.getSticky() && !inv.isAutoCraftingInventory()) break;`

Reply:
This would work for all cases where a crafting step makes the exact or lower amount than what is required for the next step. However if the craft creates for example 2 items and only 1 is required for the next step the crafting cache will only store the 1 item. The second item is supposed to be stored into a normal inventory but since `stickyInventoryFound` would be `true` in this case it would never be stored. The leftover item would only be stored in the next attempt to inject it - presumably the next tick - since it wouldn't count as sticky then.

This could work by using `if (!stickyInventoryFound && inv.getSticky()) stickyInventoryFound = true;` to set the sticky variable however, I'm going to check that when I get home tomorrow.

Reply:
Though, I think this movement is the source of a new crash on N588 https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16978

Reply:
Yes, the error is caused by the changes in the NetworkInventoryHandler since I missed readding a check for null input which I removed here. I do wonder why the import bus is doing all these calculations for a null input though...

Reply:
The second pass wasn't necessary for items but for fluids. The `FluidCraftingInventory` in ae2fc did not return the stack as prioritized. I changed this in the ae2fc PR instead, making it possible to keep the old loop here.